src/hazelweb/util/JSUtil.re

```diff
@@ -235,22 +235,18 @@ let force_get_elem_by_cls = cls =>
   | [elem, ..._] => elem
   };
 
-let update_mouse_position = evt => {
-  State.mouse_position :=
-    {
-      x: Js.Optdef.get(evt##.pageX, () => assert(false)),
-      y: Js.Optdef.get(evt##.pageY, () => assert(false)),
-    };
+let get_mouse_position = evt => {
+  (
+    Js.Optdef.get(evt##.pageX, () => assert(false)),
+    Js.Optdef.get(evt##.pageY, () => assert(false)),
+  );
 };
 
-let unsafe_get_elt_under_mouse = () => {
+let element_from_point = (x: int, y: int) => {
```

Comment:
instead of taking x and y separately, have it take an argument of type `MousePosition.t` (see above)